implementation/src/main/java/io/smallrye/mutiny/operators/uni/builders/UniJoinFirst.java

```diff
@@ -29,6 +28,7 @@ public UniJoinFirst(List<Uni<T>> unis, Mode mode, int concurrency) {
         this.unis = unis;
         this.mode = mode;
         this.concurrency = concurrency;
+        assert (mode == Mode.FIRST_WITH_ITEM || (mode == Mode.FIRST_TO_EMIT && concurrency == -1)); // Invariant enforced by the caller DSL
```

Comment:
Note: Maven Surefire enables assertions by default. It's really an invariant worth checking in tests, not at runtime.